src/connection.rs

```diff
@@ -1606,7 +1708,12 @@ async fn server_req_handler<S: AsyncRead + AsyncWrite>(
         {
             Ok(Some(ret)) => ret,
             Ok(None) => return Ok(false), // no request
-            Err(e) => return Err(e),
+            Err(e) => {
```

Comment:
what about handling err resp in other funcs such as `server_req_read_header_and_body`

Reply:
ah, that function is called by `server_req_respond`, and so any error from it will bubble up. basically, the `*_respond` functions (`server_req_respond` and `server_stream_respond`) process a request all the way up to the moment of preparing a successful response header, such that handling their return value is the only place we need to worry about sending errors